code/modules/reagents/chemistry/reagents/drink_reagents.dm

```diff
@@ -576,10 +576,17 @@
 	glass_name = "soy latte"
 	glass_desc = "A nice and refreshing beverage while you're reading."
 
+/datum/reagent/consumable/soy_latte/on_mob_add(mob/living/M)
+	M.add_trait(TRAIT_SLEEPIMMUNE)
+	..()
+
+/datum/reagent/consumable/soy_latte/on_mob_delete(mob/living/M)
+	M.remove_trait(TRAIT_SLEEPIMMUNE)
```

Comment:
But what if this is added by a different effect 🤔

Reply:
The proc needs an unique id as a second argument exactly because of that